core/ClassInfo.php

```diff
@@ -102,7 +105,14 @@ public static function dataClassesFor($class) {
 	 * @return string
 	 */
 	public static function baseDataClass($class) {
-		if (is_object($class)) $class = get_class($class);
+		if (is_object($class)) {
+			$class = get_class($class);
+		}
+		else {
+			//normalise the class case
+			$reflection = new ReflectionClass($class);
```

Comment:
I would put this into a separate method, rather than inlining the name normalisation here. That would probably require it to go into 3.2 anyway, however (maybe?). It would be better to have a single place where class normalisation is done, and it may as well be in this class anyway.